src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,54 @@
+@import "../utils/variables";
+@import "../utils/mixins";
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: $gap-size;
+  width: 7 * $day-size + 6 * $gap-size;
+  padding: 10px;
+
+  @for $day from 28 through 30 {
+    &--month-length--#{$day} :nth-child(n+#{$day+1}) {
+      display: none;
+    }
+  }
+
+  $days-of-week:
+    "mon" 0,
+    "tue" 1,
+    "wed" 2,
+    "thu" 3,
+    "fri" 4,
+    "sat" 5,
+    "sun" 6;
+
+  @each $start-day, $start-day-value in $days-of-week {
+    &--start-day--#{$start-day} :first-child {
+      margin-left: $start-day-value * ($day-size + $gap-size);
+    }
+  }
+}
+
+.calendar__cell {
+  @include day-size($day-size);
+  box-sizing: border-box;
+  background-color: #eee;
+  border: 1px solid #000;
+  display: flex;
+
+  @for $day-number from 1 through 31 {
+    &:nth-child(#{$day-number}):before {
+
+      content: "#{$day-number}";
+      margin: auto;
+    }
+  }
+
+  &:hover {
+    cursor: pointer;
+    transform: translateY(-20px);
+    animation-duration: $animation-duration;
```

Comment:
I am not sure, but for this better use property "transition" and don't put it under the hover pseudoclass, just add it for calendar__cell, like 
transition: transform $animation-suration

Reply:
I agree about the transition, but I need to leave the [transform](transform: translateY(-20px);) in the hover to set the direction) thanks a lot 🌷